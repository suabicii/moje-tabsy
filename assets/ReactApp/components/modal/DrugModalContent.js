import React from "react";

function DrugModalContent(props) {
    return (
        <>
            <button className="btn btn-close" onClick={() => {
                props.setIsModalOpen(false);
            }}>
            </button>
            <h3 className="mt-2 text-center">Edytuj</h3>
            <form onSubmit={e => {
                e.preventDefault();
            }}>
                <div className="row">
                    <div className="col-md-6">
                        <div className="form-floating mt-3">
                            <input type="text" className="form-control" id="name" placeholder="Nazwa"/>
                            <label htmlFor="name">Nazwa</label>
                        </div>
                        <div className="form-floating mt-3">
                            <input type="text" className="form-control" id="unit" placeholder="Jednostka"/>
                            <label htmlFor="unit">Jednostka (szt., ml. itp.)</label>
                        </div>
                    </div>
                    <div className="col-md-6">
                        <div className="form-floating mt-3">
                            <input type="number" step="1" className="form-control" id="quantity" placeholder="Bieżąca ilość"/>
                            <label htmlFor="quantity">Bieżąca ilość</label>
                        </div>
                        <div className="form-floating mt-3">
                            <input type="number" step="1" className="form-control" id="quantityMax"
                                   placeholder="Ilość całkowita"/>
                            <label htmlFor="quantityMax">Ilość całkowita</label>
                        </div>
                    </div>
                </div>
                <div className="form-floating mt-3">
                    <input type="number" step="1" className="form-control" id="dosing"
                           placeholder="Dawkowanie (ile razy dziennie)" value="1" aria-valuemin="1"/>
                    <label htmlFor="dosing">Dawkowanie (ile razy dziennie)</label>
                </div>
                <div className="dosing-hours mt-3">
                    <h4 className="text-center">Godziny przyjęcia dawki</h4>
                    <div className="input-group mt-3">
                        <span className="input-group-text"><i
                            className="fa-solid fa-clock"></i></span>
                        <input type="time" name="hour1" className="form-control" aria-label="Godzina przyjęcia dawki"/>
                    </div>
                </div>
                <div className="d-flex justify-content-between mt-3">
                    <button type="submit" className="btn btn-info px-5">Zapisz</button>
                    <button type="button" className="btn btn-outline-dark px-5" onClick={() => {
                        props.setIsModalOpen(false);
                    }}>
                        Anuluj
                    </button>
                </div>
            </form>
        </>
    );
}

export default DrugModalContent;