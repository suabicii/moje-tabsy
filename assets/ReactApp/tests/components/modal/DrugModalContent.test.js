import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import DrugModalContent from "../../../components/modal/DrugModalContent";

it('should correctly render modal content with form', function () {
    const renderer = new ReactShallowRenderer();
    const setIsModalOpen = jest.fn();
    renderer.render(<DrugModalContent setIsModalOpen={setIsModalOpen}/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
}); 