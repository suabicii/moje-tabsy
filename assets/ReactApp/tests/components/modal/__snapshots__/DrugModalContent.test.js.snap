// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render modal content with form 1`] = `
<React.Fragment>
  <button
    className="btn btn-close"
    onClick={[Function]}
  />
  <h3
    className="mt-2 text-center"
  >
    Edytuj
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <div
          className="form-floating mt-3"
        >
          <input
            className="form-control"
            id="name"
            placeholder="Nazwa"
            type="text"
          />
          <label
            htmlFor="name"
          >
            Nazwa
          </label>
        </div>
        <div
          className="form-floating mt-3"
        >
          <input
            className="form-control"
            id="unit"
            placeholder="Jednostka"
            type="text"
          />
          <label
            htmlFor="unit"
          >
            Jednostka (szt., ml. itp.)
          </label>
        </div>
      </div>
      <div
        className="col-md-6"
      >
        <div
          className="form-floating mt-3"
        >
          <input
            className="form-control"
            id="quantity"
            placeholder="Bieżąca ilość"
            step="1"
            type="number"
          />
          <label
            htmlFor="quantity"
          >
            Bieżąca ilość
          </label>
        </div>
        <div
          className="form-floating mt-3"
        >
          <input
            className="form-control"
            id="quantityMax"
            placeholder="Ilość całkowita"
            step="1"
            type="number"
          />
          <label
            htmlFor="quantityMax"
          >
            Ilość całkowita
          </label>
        </div>
      </div>
    </div>
    <div
      className="form-floating mt-3"
    >
      <input
        aria-valuemin="1"
        className="form-control"
        id="dosing"
        placeholder="Dawkowanie (ile razy dziennie)"
        step="1"
        type="number"
        value="1"
      />
      <label
        htmlFor="dosing"
      >
        Dawkowanie (ile razy dziennie)
      </label>
    </div>
    <div
      className="dosing-hours mt-3"
    >
      <h4
        className="text-center"
      >
        Godziny przyjęcia dawki
      </h4>
      <div
        className="input-group mt-3"
      >
        <span
          className="input-group-text"
        >
          <i
            className="fa-solid fa-clock"
          />
        </span>
        <input
          aria-label="Godzina przyjęcia dawki"
          className="form-control"
          name="hour1"
          type="time"
        />
      </div>
    </div>
    <div
      className="d-flex justify-content-between mt-3"
    >
      <button
        className="btn btn-info px-5"
        type="submit"
      >
        Zapisz
      </button>
      <button
        className="btn btn-outline-dark px-5"
        onClick={[Function]}
        type="button"
      >
        Anuluj
      </button>
    </div>
  </form>
</React.Fragment>
`;
